lib/Backend/GlobOpt.cpp

```diff
@@ -470,6 +470,7 @@ GlobOpt::OptBlock(BasicBlock *block)
     PrepareLoopArrayCheckHoist();
 
     block->MergePredBlocksValueMaps(this);
+    this->currentBlock->PathDepBranchFolding(this);
```

Comment:
nit: if `block` and `this->currentBlock` are the same here, we should try to be consistent with which one we use.